### PR TITLE
Feature/chips for store types

### LINKED ISSUE #108: Remove Live-Tracking for Now and Improve Display of Selected Store Type
* Do not remove the whole code for the live tracking filtering, but remove UI as well as the filtering in the map_bloc
* display a Chip (https://www.material.io/components/chips/) with the selected store type and label

### PR DESCRIPTION
**Pull request by LukkyLukke**
Filter dialog has been removed and replaced with chips placed directly on the map (easily accessible)
closes #108

### CODE REVIEW

File: lib/ui/map/map_view.dart [hunk 1]
```diff
@@ -121,64 +122,66 @@ class MapViewState extends State<MapView> {
...
-                BlocProvider.of<MapBloc>(context).state is MapLoading) {
-              return;
-            }
-
-            _fetchLocationsIfNeeded(currentCameraPosition);
-          },
-          markers: Set<Marker>.of(clusteredMarkers),
+
+
+//            Try out chips
```
Comment (juliaringler) [inline lib/ui/map/map_view.dart]:
```suggestion

```

File: lib/ui/map/map_chips.dart [hunk 2]
```diff
@@ -0,0 +1,90 @@
+
+
+import 'package:reservation_system_customer/ui_imports.dart';
+
+class MapChips extends StatefulWidget {
+  final MapBloc mapBloc;
+
+  const MapChips({Key key, this.mapBloc}) : super(key: key);
```
Comment (juliaringler) [inline lib/ui/map/map_chips.dart]: do not inject the bloc here, it is already in the build context

File: lib/ui/map/map_chips.dart [hunk 3]
```diff
@@ -0,0 +1,90 @@
+
+
+import 'package:reservation_system_customer/ui_imports.dart';
+
+class MapChips extends StatefulWidget {
```
Comment (juliaringler) [inline lib/ui/map/map_chips.dart]: make stateful

File: lib/ui/map/map_chips.dart [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+
+  @override
+  void initState() {
+    selectedType = widget.mapBloc.state.filterSettings.locationType;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Wrap(
```
Comment (juliaringler) [inline lib/ui/map/map_chips.dart]:
```suggestion
    return BlocBuilder<MapBloc, MapState>(
      condition: (oldState, newState) => oldState.filterSettings != newState.filterSettings,
      builder: (context, state) {
       return Wrap(....)
       };
      
```

File: lib/ui/map/map_chips.dart [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  _MapChipsState createState() => _MapChipsState();
+}
+
+class _MapChipsState extends State<MapChips> {
+  LocationType selectedType;
+
+
+  @override
+  void initState() {
+    selectedType = widget.mapBloc.state.filterSettings.locationType;
```
Comment (juliaringler) [inline lib/ui/map/map_chips.dart]: no need for this when you use the blocbuilder below

File: lib/ui/map/map_view.dart [hunk 6]
```diff
@@ -40,6 +39,8 @@ class MapViewState extends State<MapView> {
   final int minZoom = 0;
   final int maxZoom = 18;
 
+  LocationType selectedType = LocationType.supermarket;
```
Comment (juliaringler) [inline lib/ui/map/map_view.dart]: can also be removed now

File: lib/ui/map/map_view.dart [hunk 7]
```diff
@@ -121,64 +122,66 @@ class MapViewState extends State<MapView> {
...
-            _fetchLocationsIfNeeded(currentCameraPosition);
-          },
-          markers: Set<Marker>.of(clusteredMarkers),
+
+
+//            Try out chips
+            SafeArea(
+              child: Align(
+                alignment: Alignment.topCenter,
+                child: MapChips(mapBloc: BlocProvider.of<MapBloc>(context),)
```
Comment (juliaringler) [inline lib/ui/map/map_view.dart]:
```suggestion
                child: MapChips(),
```

File: lib/ui/map/map_chips.dart [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+        return Padding(
+          padding: EdgeInsets.all(0),
+          child: ChoiceChip(
+            label: Text(
+                l.localized(context),
+                style: TextStyle(
+                  color: Colors.black,
+                )
+            ),
+            selected: selectedType == l,
```
Comment (juliaringler) [inline lib/ui/map/map_chips.dart]:
```suggestion
            selected: state.filterSettings.locationType == l,
```

File: lib/ui/map/map_chips.dart (hunk 3 above)
Comment (LukkyLukke) [reply lib/ui/map/map_chips.dart]: do you mean stateless?

**Review by juliaringler — APPROVED** [review]

**Pull request merged**